pilot/cmd/pilot-agent/status/server_test.go

```diff
@@ -556,48 +567,100 @@ func TestAppProbe(t *testing.T) {
+		{
+			probePath: "app-health/remote-redirect/livez",
+			config: KubeAppProbers{
+				"/app-health/remote-redirect/livez": &Prober{
+					HTTPGet: &apimirror.HTTPGetAction{
+						Path: "remote-redirect",
+						Port: intstr.IntOrString{IntVal: int32(appPort)},
+					},
+				},
+			},
+			statusCode: http.StatusOK,
+		},
 	}
-	for _, tc := range testCases {
-		t.Run(tc.probePath, func(t *testing.T) {
-			appProber, err := json.Marshal(tc.config)
-			if err != nil {
-				t.Fatalf("invalid app probers")
-			}
-			config := Options{
-				StatusPort:     0,
-				KubeAppProbers: string(appProber),
-			}
-			// Starts the pilot agent status server.
-			server, err := NewServer(config)
-			if err != nil {
-				t.Fatalf("failed to create status server %v", err)
-			}
-			ctx, cancel := context.WithCancel(context.Background())
-			defer cancel()
-			go server.Run(ctx)
-
-			var statusPort uint16
-			for statusPort == 0 {
-				server.mutex.RLock()
-				statusPort = server.statusPort
-				server.mutex.RUnlock()
-			}
+	testFn := func(t *testing.T, tc test) {
+		appProber, err := json.Marshal(tc.config)
+		if err != nil {
+			t.Fatalf("invalid app probers")
+		}
+		config := Options{
+			StatusPort:     0,
+			KubeAppProbers: string(appProber),
+		}
+		// Starts the pilot agent status server.
+		server, err := NewServer(config)
+		if err != nil {
+			t.Fatalf("failed to create status server %v", err)
+		}
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		go server.Run(ctx)
+
+		var statusPort uint16
+		for statusPort == 0 {
+			server.mutex.RLock()
+			statusPort = server.statusPort
+			server.mutex.RUnlock()
+		}
 
-			client := http.Client{}
-			req, err := http.NewRequest("GET", fmt.Sprintf("http://localhost:%v/%s", statusPort, tc.probePath), nil)
-			if err != nil {
-				t.Fatalf("[%v] failed to create request", tc.probePath)
-			}
-			resp, err := client.Do(req)
-			if err != nil {
-				t.Fatal("request failed: ", err)
-			}
-			defer resp.Body.Close()
-			if resp.StatusCode != tc.statusCode {
-				t.Errorf("[%v] unexpected status code, want = %v, got = %v", tc.probePath, tc.statusCode, resp.StatusCode)
-			}
-		})
+		client := http.Client{}
+		req, err := http.NewRequest("GET", fmt.Sprintf("http://localhost:%v/%s", statusPort, tc.probePath), nil)
+		if err != nil {
+			t.Fatalf("[%v] failed to create request", tc.probePath)
+		}
+		resp, err := client.Do(req)
+		if err != nil {
+			t.Fatal("request failed: ", err)
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode != tc.statusCode {
+			t.Errorf("[%v] unexpected status code, want = %v, got = %v", tc.probePath, tc.statusCode, resp.StatusCode)
+		}
 	}
+	for _, tc := range testCases {
+		t.Run(tc.probePath, func(t *testing.T) { testFn(t, tc) })
+	}
+	// Next we check ever
+	t.Run("status codes", func(t *testing.T) {
+		for code := http.StatusOK; code <= http.StatusNetworkAuthenticationRequired; code++ {
+			if http.StatusText(code) == "" { // Not a valid HTTP code
```

Comment:
same as code=0?

Reply:
No, this makes sure that a code like `399` which is not defined is not tested.